pkg/runtime/unstructured.go

```diff
@@ -28,7 +28,9 @@ import (
 // type, which can be used for generic access to objects without a predefined scheme.
 var UnstructuredJSONScheme ObjectDecoder = unstructuredJSONScheme{}
 
-type unstructuredJSONScheme struct{}
+type unstructuredJSONScheme struct {
+	Decoder
```

Comment:
Decoder is an interface, and it will be nil, so this will crash if anyone tries to call the new method. Please add a dummy DecodeIntoWithDefaultVersionKind instead.